src/conf_mode/system-logs.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import exit
+
+from vyos import ConfigError
+from vyos import airbag
+from vyos.config import Config
+from vyos.logger import syslog
+from vyos.template import render_to_string
+from vyos.util import read_file, write_file
+airbag.enable()
+
+# path to logrotate config for atop
+logrotate_atop_file = '/etc/logrotate.d/vyos-atop'
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base = ['system', 'logs']
+    logs_config = conf.get_config_dict(base)
+
+    return logs_config
+
+
+def verify(logs_config):
+    # Nothing to verify here
+    pass
+
+
+def generate(logs_config):
+    # get configuration for logrotate atop
+    logrotate_atop = logs_config.get('logs', {}).get('logrotate',
+                                                     {}).get('atop', {})
+    # read current config file for atop
+    logrotate_atop_current = read_file(logrotate_atop_file)
+    # generate new config file for atop
+    logrotate_atop_new = render_to_string('logs/logrotate/vyos-atop.tmpl',
+                                          logrotate_atop)
+    # update configuration files if this is necessary
+    if logrotate_atop_new != logrotate_atop_current:
```

Comment:
Why is this delta check mandatory? I would always write this file - I see no harm in this as it does not happen 1000 times a second. Then you can simply use `render()` which writes to a a file.

Reply:
I think that blindly overwriting files with each config change is a bad design. If they are located in the `/run` this may be acceptable, but when they are in permanent storage this seems to be strange.
Maybe, we should even add an optional ability to the `util.write_file()` that can do this check, to avoid extra logic in config scripts.

Reply:
Well, system startup cleans files our of /etc to not have dead files lurking around. In my opinion always generating the required config files is a good design and we should not leave that path as we do it for every other part of the system. ALso delta checks tend to just add additional problems as corner cases are missed.

Reply:
The checking was removed.